portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.js

```diff
@@ -1,21 +1,45 @@
+const moment = require('moment');
 const { extractCsvData, removeCellAddressesFromArray } = require('../../../utils/csv-utils');
 const { validateCsvData } = require('./utilisation-report-validator');
-const { getCurrentReportDueDate, getCurrentReportPeriod } = require('./utilisation-report-status');
+const { getCurrentReportDueDate } = require('./utilisation-report-status');
 const api = require('../../../api');
 
 const getUtilisationReportUpload = async (req, res) => {
   const { user, userToken } = req.session;
+  const bankId = user.bank.id;
   try {
-    const currentDate = new Date();
-    const firstDayOfCurrentMonth = new Date(currentDate.getFullYear(), currentDate.getMonth(), 1);
-    const { reportPeriod, month, year } = getCurrentReportPeriod(firstDayOfCurrentMonth);
+    const dueReports = await api.getDueReportsByBank(userToken, bankId);
+    if (dueReports.length === 0) {
+      // TODO: feat(FN-1089) - result when reports are up to date
+      return res.render('utilisation-report-service/utilisation-report-upload/utilisation-report-upload.njk', {
+        user,
+        primaryNav: 'utilisation_report_upload',
+        reportPeriod: 'reportPeriod',
+        reportDueDate: 'reportDueDate',
+      });
+    }
+
+    const { year, month } = dueReports.at(-1);
+    const reportPeriod = `${month} ${year}`;
+    const reportDate = moment(reportPeriod, 'MMMM yyyy');
```

Comment:
I see we have both `moment` and `date-fns` in the `portal` package. For new work I think we should prefer to use `date-fns` as `moment` has now been deprecated. This will also help to keep things more consistent.

Reply:
This has been removed :)